app/cdap/testids/TestidsProvider.tsx

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright © 2016-2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { useContext } from 'react';
+import _get from 'lodash/get';
+import testids from './testids.yaml';
+import { DataTestIdGetter, TestidContext } from './TestidContext';
+
+interface ITestidProviderProps {
+  children: React.ReactNode;
+}
+
+function getDataTestidInternal(prefixPath: string, siblingKey?: string | number): string {
+  const actualTestidPrefix = _get(testids, prefixPath);
+  if (actualTestidPrefix !== null && typeof actualTestidPrefix !== 'string') {
```

Comment:
Can you please brief about this condition, am little unclear what it is doing .

Reply:
done.